### PR TITLE
Better structure of the Ophyd classes

### PR DESCRIPTION
**Pull request by mrakitin**
WIP

### CODE REVIEW

File: src/hextools/germ/caproto_ioc.py [hunk 1]
```diff
@@ -96,64 +220,73 @@ async def stage(self, instance, value):
...
+                data=np.full(fill_value=np.nan, shape=(1, *frame_shape)),
+                maxshape=(None, *frame_shape),
+                chunks=(1, *frame_shape),
                 dtype="float32",
             )
             self._h5file_desc.swmr_mode = True
             return True
 
-        elif value == "unstaged":
+        if value == StageStates.UNSTAGED.value:
```
Comment (tacaswell) [inline src/hextools/germ/caproto_ioc.py:234]: in this block also reset all of the various internal state to `None`.

It won't "fix" the problem, but will at least convert the errors to ones that are clearer.

**Review by jwlodek — APPROVED** [review]
LGTM!

**Comment by mrakitin** [comment]
There was an issue with saving the data with 30s+ counts, I am working on resolving it.

**Comment by mrakitin** [comment]
I will self-merge to get ready for production.

**Pull request merged**